### PR TITLE
Fix code style

### PR DESCRIPTION
**Pull request by valentynpatsera**
This PR fixes the following code style issues:

- indentation in some places
- replaces `let` with `const` where possible
- replaces double brackets `" "` with single brackets `' '` to keep only one type of brackets in the code base
- declares an undeclared variable
- inverts `if` conditions to reduce nesting
- removes redundant `else` statement

### CODE REVIEW

File: scripts/todo.js [hunk 1]
```diff
@@ -256,41 +250,38 @@ window.onload = function() {
...
+      // The JavaScript date object creates month values as a number between 0 and 11.
+      const monthNumber = MONTHS.indexOf(cursor.value.month);
+      if (monthNumber === -1) alert('Incorrect month entered in database.');
+      
+      // check if the current hours, minutes, day, month and year values match the stored values for each task in the IDB.
+      // The + operator in this case converts numbers with leading zeros into their non leading zero equivalents, so e.g.
+      // 09 -> 9. This is needed because JS date number values never have leading zeros, but our data might.
+      // The secondsCheck = 0 check is so that you don't get duplicate notifications for the same task. The notification
+      // will only appear when the seconds is 0, meaning that you won't get more than one notification for each task
+      if(+(cursor.value.hours) == hourCheck && +(cursor.value.minutes) == minuteCheck && +(cursor.value.day) == dayCheck && monthNumber == monthCheck && cursor.value.year == yearCheck && cursor.value.notified == 'no') {
```
Comment (tshemsedinov) [inline scripts/todo.js:268]:
```suggestion
      const { hours, minutes, day, year } = cursor.value;

      let mached = parseInt(hours) === hourCheck;
      mached &&= parseInt(minutes) === minuteCheck;
      mached &&= parseInt(day) === dayCheck;
      mached &&= parseInt(monthNumber) === monthCheck;
      mached &&= parseInt(year) === yesrCheck;

      if (mached && notified === 'no') {
```

File: scripts/todo.js [hunk 2]
```diff
@@ -256,41 +250,38 @@ window.onload = function() {
...
+
+        // If the numbers all do match, run the createNotification() function to create a system notification
+        // but only if the permission is set
+
+        if(Notification.permission === 'granted') {
```
Comment (tshemsedinov) [inline scripts/todo.js:273]:
```suggestion
        if (Notification.permission === 'granted') {
```

File: scripts/todo.js [hunk 3]
```diff
@@ -309,8 +300,8 @@ window.onload = function() {
     }
 
     // Let's check if the browser supports notifications
-    if (!"Notification" in window) {
-      console.log("This browser does not support notifications.");
+    if (!'Notification' in window) {
```
Comment (tshemsedinov) [inline scripts/todo.js:303]:
```suggestion
    if (!Reflect.has(window, 'Notification')) {
```

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
Thanks for your contribution! See suggestions below how to improve code style deeply

**Comment by valentynpatsera** [comment]
@tshemsedinov, thank you! I thought such changes should be implemented in a separate PR. 
@tshemsedinov, @wbamberg, is it OK to do the suggested and also other similar changes in this PR?

**Comment by tshemsedinov** [comment]
@valentynpatsera I'd prefer to have all code style fixes (I think all this fixes are code-style, because they do not add/remove functionality and don't related to performance optimizations or other aspects) in a single PR to see everything together.

File: scripts/todo.js [hunk 4]
```diff
@@ -103,55 +103,54 @@ window.onload = function() {
...
+        daySuffix = 'nd';
+      } else if(cursor.value.day == 3 || cursor.value.day == 23) {
+        daySuffix = 'rd';
+      } else {
+        daySuffix = 'th';
+      }
+
+      // build the to-do list entry and put it into the list item.
+      const toDoText = `${cursor.value.taskTitle} — ${cursor.value.hours}:${cursor.value.minutes}, ${cursor.value.month} ${cursor.value.day}${daySuffix}  ${cursor.value.year}.`;
+      const listItem = createListItem(toDoText);
```
[the remark this answers is not in the document]
Comment (wbamberg) [reply scripts/todo.js:129]: I keep looking at this, and thinking: is there a reason we're not using [`<datetime-local>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local) to enter the due date, storing it as a single `Date` value, and using [`Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/format) to display it?

I appreciate we might want to separate "style" from "substance" but I think these changes go beyond "style" if we might want to make deeper changes anyway in future PRs?

**Comment by valentynpatsera** [comment]
@tshemsedinov, as far as I understand there is no objection from @wbamberg to keep the suggested changes in this PR, so I will add those and maybe a couple more. Thanks.

File: scripts/todo.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tshemsedinov) [reply scripts/todo.js:129]: @wbamberg I think @valentynpatsera may finish this style changes with my minor help but not concentration on time-consuming refactoring like extracting functions and so on, after that we'd ask him (if he wants and can, of course) to implement your great ideas and if we need more developers I can invite them here from community. I promise to review the code even under the bombing sitting in shelter in Ukraine if only i had electricity and internet.

**Comment by tshemsedinov** [comment]
@valentynpatsera see my reply https://github.com/mdn/to-do-notifications/pull/21#discussion_r830512613
So you can count on my help. I hope you are in a safe place now.

**Comment by valentynpatsera** [comment]
@tshemsedinov, thank you! 
Yes, I can work, I am in Europe. I know you are in Ukraine, stay safe.

File: scripts/todo.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (wbamberg) [reply scripts/todo.js]: So sorry @tshemsedinov for the awful situation you and your people are in.

**Review by tshemsedinov — APPROVED** [review]
LGTM, but change `==` to `===` in a few places as a basic style fixes. All other stuff like `ordinal(day: number): string` and using `input` and `Intl` we may move to another PR.

**Comment by valentynpatsera** [comment]
@tshemsedinov, @wbamberg, all suggested changes except a few small ones have been added. Please review.
I will make separate PRs for the `input` and `Intl`, extracting functions, `Notifications` and code duplicates refactoring.

**Comment by valentynpatsera** [comment]
@wbamberg , could we get this one merged in so I could keep on with the rest of the updates?

File: scripts/todo.js [hunk 5]
```diff
@@ -40,17 +40,17 @@ window.onload = function() {
   note.appendChild(createListItem('App initialised.'));
   // In the following line, you should include the prefixes of implementations you want to test.
   window.indexedDB = window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB;
-  // DON'T use "var indexedDB = ..." if you're not in a function.
+  // DON'T use 'var indexedDB = ...' if you're not in a function.
   // Moreover, you may need references to some window.IDB* objects:
   window.IDBTransaction = window.IDBTransaction || window.webkitIDBTransaction || window.msIDBTransaction;
   window.IDBKeyRange = window.IDBKeyRange || window.webkitIDBKeyRange || window.msIDBKeyRange;
   // (Mozilla has never prefixed these objects, so we don't need window.mozIDB*)
```
Comment (teoli2003) [inline scripts/todo.js:47]: Ditto.
```suggestion
```

File: scripts/todo.js [hunk 6]
```diff
@@ -40,17 +40,17 @@ window.onload = function() {
...
 
   // Let us open our database
-  const DBOpenRequest = window.indexedDB.open("toDoList", 4);
+  const DBOpenRequest = window.indexedDB.open('toDoList', 4);
 
   // Gecko-only IndexedDB temp storage option:
-  // var request = window.indexedDB.open("toDoList", {version: 4, storage: "temporary"});
+  // var request = window.indexedDB.open('toDoList', {version: 4, storage: 'temporary'});
```
Comment (teoli2003) [inline scripts/todo.js:53]: Let's be coherent:
```suggestion
  // const DBOpentRequest = window.indexedDB.open('toDoList', {version: 4, storage: 'temporary'});
```

File: scripts/todo.js [hunk 7]
```diff
@@ -40,17 +40,17 @@ window.onload = function() {
...
 
   // these two event handlers act on the database being opened successfully, or not
   DBOpenRequest.onerror = function(event) {
```
Comment (teoli2003) [inline scripts/todo.js:56]: Let's use arrow functions for anonymous functions where possible.
```suggestion
  DBOpenRequest.onerror = (event) => {
```

Comment (teoli2003) [reply scripts/todo.js:56]: Please do the same on line 60 (GitHub UI doesn't allow me to make a suggestion there 😢 )

File: scripts/todo.js [hunk 8]
```diff
@@ -72,25 +72,25 @@ window.onload = function() {
   // window.indexedDB.open line above
   //it is only implemented in recent browsers
   DBOpenRequest.onupgradeneeded = function(event) {
```
Comment (teoli2003) [inline scripts/todo.js:74]:
```suggestion
  DBOpenRequest.onupgradeneeded = (event) => {
```

File: scripts/todo.js [hunk 9]
```diff
@@ -72,25 +72,25 @@ window.onload = function() {
...
-    let db = event.target.result;
+    const db = event.target.result;
```
Comment (teoli2003) [inline scripts/todo.js:75]: We have `let db` near the top, I think it should be:
```suggestion
    db = event.target.result;
```

File: scripts/todo.js [hunk 10]
```diff
@@ -72,25 +72,25 @@ window.onload = function() {
...
 
     db.onerror = function(event) {
```
Comment (teoli2003) [inline scripts/todo.js:77]:
```suggestion
    db.onerror = (event) => {
```

File: scripts/todo.js [hunk 11]
```diff
@@ -103,128 +103,114 @@ window.onload = function() {
     }
 
     // Open our object store and then get a cursor list of all the different data items in the IDB to iterate through
-    let objectStore = db.transaction('toDoList').objectStore('toDoList');
+    const objectStore = db.transaction('toDoList').objectStore('toDoList');
     objectStore.openCursor().onsuccess = function(event) {
```
Comment (teoli2003) [inline scripts/todo.js:107]:
```suggestion
    objectStore.openCursor().onsuccess = (event) => {
```

File: scripts/todo.js [hunk 12]
```diff
@@ -103,128 +103,114 @@ window.onload = function() {
...
+      taskList.appendChild(listItem);
+
+      // create a delete button inside each list item, giving it an event handler so that it runs the deleteButton()
+      // function when clicked
+      const deleteButton = document.createElement('button');
+      listItem.appendChild(deleteButton);
+      deleteButton.textContent = 'X';
+      // here we are setting a data attribute on our delete button to say what task we want deleted if it is clicked!
+      deleteButton.setAttribute('data-task', taskTitle);
+      deleteButton.onclick = function(event) {
```
Comment (teoli2003) [inline scripts/todo.js:138]:
```suggestion
      deleteButton.onclick = (event) => {
```

File: scripts/todo.js [hunk 13]
```diff
@@ -103,128 +103,114 @@ window.onload = function() {
...
-        displayData();
-      };
+    // open a read/write db transaction, ready for adding the data
+    const transaction = db.transaction(['toDoList'], 'readwrite');
 
-      transaction.onerror = function() {
-        note.appendChild(createListItem(`Transaction not opened due to error: ${transaction.error}`));
-      };
+    // report on the success of the transaction completing, when everything is done
+    transaction.oncomplete = function() {
```
Comment (teoli2003) [inline scripts/todo.js:169]:
```suggestion
    transaction.oncomplete = () => {
```

File: scripts/todo.js [hunk 14]
```diff
@@ -103,128 +103,114 @@ window.onload = function() {
...
-          month.value = 'January';
-          year.value = 2020;
-
-        };
+      // update the display of data to show the newly added item, by running displayData() again.
+      displayData();
+    };
 
-      };
+    transaction.onerror = function() {
```
Comment (teoli2003) [inline scripts/todo.js:176]:
```suggestion
    transaction.onerror = () => {
```

File: scripts/todo.js [hunk 15]
```diff
@@ -103,128 +103,114 @@ window.onload = function() {
...
+    const objectStore = transaction.objectStore('toDoList');
+    console.log(objectStore.indexNames);
+    console.log(objectStore.keyPath);
+    console.log(objectStore.name);
+    console.log(objectStore.transaction);
+    console.log(objectStore.autoIncrement);
+
+    // Make a request to add our newItem object to the object store
+    const objectStoreRequest = objectStore.add(newItem[0]);
+    objectStoreRequest.onsuccess = function(event) {
```
Comment (teoli2003) [inline scripts/todo.js:190]:
```suggestion
    objectStoreRequest.onsuccess = (event) => {
```

File: scripts/todo.js [hunk 16]
```diff
@@ -103,128 +103,114 @@ window.onload = function() {
...
+    const dataTask = event.target.getAttribute('data-task');
 
     // open a database transaction and delete the task, finding it by the name we retrieved above
-    let transaction = db.transaction(["toDoList"], "readwrite");
-    let request = transaction.objectStore("toDoList").delete(dataTask);
+    const transaction = db.transaction(['toDoList'], 'readwrite');
+    transaction.objectStore('toDoList').delete(dataTask);
 
     // report that the data item has been deleted
     transaction.oncomplete = function() {
```
Comment (teoli2003) [inline scripts/todo.js:216]:
```suggestion
    transaction.oncomplete = () => {
```

File: scripts/todo.js [hunk 17]
```diff
@@ -256,74 +242,76 @@ window.onload = function() {
...
 
-  // askNotificationPermission function to ask for permission when the "Enable notifications" button is clicked
+  // askNotificationPermission function to ask for permission when the 'Enable notifications' button is clicked
 
   function askNotificationPermission() {
     // function to actually ask the permissions
     function handlePermission(permission) {
       // Whatever the user answers, we make sure Chrome stores the information
-      if(!('permission' in Notification)) {
+      if (!Reflect.has(Notification, 'permission')) {
```
Comment (teoli2003) [inline scripts/todo.js:287]: This is likely correct, but I don't know the advantages here.

File: scripts/todo.js [hunk 18]
```diff
@@ -256,74 +242,76 @@ window.onload = function() {
...
     // Let's check if the browser supports notifications
-    if (!"Notification" in window) {
-      console.log("This browser does not support notifications.");
+    if (!Reflect.has(window, 'Notification')) {
+      console.log('This browser does not support notifications.');
     } else {
-      if(checkNotificationPromise()) {
+      if (checkNotificationPromise()) {
         Notification.requestPermission()
         .then((permission) => {
```
Comment (teoli2003) [inline scripts/todo.js:305]: Please indent here.

File: scripts/todo.js [hunk 19]
```diff
@@ -256,74 +242,76 @@ window.onload = function() {
...
           handlePermission(permission);
-        })
+        });
       } else {
         Notification.requestPermission(function(permission) {
```
Comment (teoli2003) [inline scripts/todo.js:309]:
```suggestion
        Notification.requestPermission((permission) => {
```

File: scripts/todo.js [hunk 20]
```diff
@@ -335,52 +323,61 @@ window.onload = function() {
...
 
     // first open up a transaction as usual
-    let objectStore = db.transaction(['toDoList'], "readwrite").objectStore('toDoList');
+    const objectStore = db.transaction(['toDoList'], 'readwrite').objectStore('toDoList');
 
     // get the to-do list object that has this title as it's title
-    let objectStoreTitleRequest = objectStore.get(title);
+    const objectStoreTitleRequest = objectStore.get(title);
 
     objectStoreTitleRequest.onsuccess = function() {
```
Comment (teoli2003) [inline scripts/todo.js:355]:
```suggestion
    objectStoreTitleRequest.onsuccess = () => {
```

File: scripts/todo.js [hunk 21]
```diff
@@ -335,52 +323,61 @@ window.onload = function() {
...
-  }
+      };
+    };
+  };
 
   // using a setInterval to run the checkDeadlines() function every second
   setInterval(checkDeadlines, 1000);
 }
+
+const ordinal = (day) => {
```
Comment (teoli2003) [inline scripts/todo.js:376]: This is not an anonymous function. Why not this?
```suggestion
function ordinal(day) {
```

File: scripts/todo.js [hunk 22]
```diff
@@ -335,52 +323,61 @@ window.onload = function() {
...
+  const n = day.toString();
+  const last = n.slice(-1);
+  if (last === '1') return `${n}st`;
+  if (last === '2') return `${n}nd`;
+  if (last === '3') return `${n}rd`;
+  return `${n}th`;
+};
```
Comment (teoli2003) [inline scripts/todo.js:383]:
```suggestion
};

```

File: scripts/todo.js [hunk 23]
```diff
@@ -8,9 +8,9 @@ const note = document.getElementById('notifications');
...
 
 // create a blank instance of the object that is used to transfer data into the IDB. This is mainly for reference
-let newItem = [
-      { taskTitle: "", hours: 0, minutes: 0, day: 0, month: "", year: 0, notified: "no" }
-    ];
+const newItem = [
+  { taskTitle: '', hours: 0, minutes: 0, day: 0, month: '', year: 0, notified: 'no' },
+];
 
 // all the variables we need for the app
```
Comment (teoli2003) [inline scripts/todo.js:15]:
```suggestion
// all the UI elements we need for the app
```

File: scripts/todo.js [hunk 24]
```diff
@@ -40,17 +40,17 @@ window.onload = function() {
   note.appendChild(createListItem('App initialised.'));
```
Comment (teoli2003) [inline scripts/todo.js:40]: On line 39:
```
window.onload = () => {
```

File: scripts/todo.js [hunk 25]
```diff
@@ -256,74 +242,76 @@ window.onload = function() {
     const yearCheck = now.getFullYear();
 
     // again, open a transaction then a cursor to iterate through all the data items in the IDB
-    let objectStore = db.transaction(['toDoList'], "readwrite").objectStore('toDoList');
+    const objectStore = db.transaction(['toDoList'], 'readwrite').objectStore('toDoList');
     objectStore.openCursor().onsuccess = function(event) {
```
Comment (teoli2003) [inline scripts/todo.js:246]:
```suggestion
    objectStore.openCursor().onsuccess = (event) => {
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
This is a great improvement, thanks! I have a few questions or details to fix here or there. Also, we should use arrow functions to define anonymous functions.

File: scripts/todo.js (hunk 25 above)
Comment (valentynpatsera) [reply scripts/todo.js:246]: Sure, I will add this to the PR description.

File: scripts/todo.js (hunk 21 above)
Comment (valentynpatsera) [reply scripts/todo.js:376]: because it is not using `this` context

**Review by teoli2003 — APPROVED** [review]
👍 Let's merge this. What are the next phases?

**Comment by valentynpatsera** [comment]
@teoli2003, I am working on code duplication refactoring as well as `Notification` issue described in this [PR](https://github.com/mdn/to-do-notifications/pull/18). Let me know if you think there is something else that needs attention first.

**Comment by teoli2003** [comment]
That's fine!

**Pull request merged**